frontend/catalyst/api_extensions/quantum_operators.py

```diff
@@ -387,19 +388,23 @@ def _callee():
 
 
 ## IMPL ##
-class MidCircuitMeasure(HybridOp):
+class MidCircuitMeasure(HybridOp, MidMeasureMP):
     """Operation representing a mid-circuit measurement."""
 
     binder = qmeasure_p.bind
 
+    def __init__(self, in_classical_tracers, out_classical_tracers, regions: List[HybridOpRegion]):
+        HybridOp.__init__(self, in_classical_tracers, out_classical_tracers, regions)
+        # Initializing MidMeasureMP attributes
+        self.wires = [in_classical_tracers[0]]
+        self.postselect = in_classical_tracers[1]
+        # TODO: Check if we need to initialize other MidMeasureMP attributes as well
+
     def trace_quantum(self, ctx, device, trace, qrp) -> QRegPromise:
         op = self
-        wire = op.in_classical_tracers[0]
-        qubit = qrp.extract([wire])[0]
-        postselect = op.in_classical_tracers[1]
-
-        qubit2 = op.bind_overwrite_classical_tracers(ctx, trace, qubit, postselect=postselect)
-        qrp.insert([wire], [qubit2])
+        qubit = qrp.extract(op.wires)[0]
+        qubit2 = op.bind_overwrite_classical_tracers(ctx, trace, qubit, postselect=op.postselect)
+        qrp.insert(op.wires, [qubit2])
```

Comment:
Not your code, but we could use `self` directly instead of `op`. Got confused about where this op came from 😅

Reply:
Yeah I was confused about that as well. I've changed it. Let me know if you think it makes sense to also change this for the other `HybridOp`s and I'll do that.

Reply:
Thanks! I think this is fine for now :)